### PR TITLE
Better kernel debugging: export TNUMs to GAP; improved gdb and lldb experience

### PR DESCRIPTION
**Pull request by fingolfin**
These tweaks have been quite helpful for me when debugging certain kernel problems.

### CODE REVIEW

**Comment by markuspf** [comment]
I have to correct myself a bit: I think I want the `Stat` and `Expr` "TNUMs" to be exported as constants, which I could do myself.

**Comment by ChrisJefferson** [comment]
When I've done this in the past, I've written one header (which isn't valid GAP code) which defines all the elements of the TNUM, by calling a macro for each, then including the header multiple times, replacing the function each time (does that make sense?)

File: src/debug.c [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+   (VoidFunc)SwitchToOldLVars,
+   (VoidFunc)SwitchToOldLVarsAndFree,
+   (VoidFunc)TEST_OBJ_FLAG,
+   (VoidFunc)TNAM_OBJ,
+   (VoidFunc)TNUM_BAG,
+   (VoidFunc)TNUM_OBJ,
+   (VoidFunc)UNB_LIST,
+   (VoidFunc)VAL_MACFLOAT,
+   (VoidFunc)VisitStatIfHooked,
+};
```
Comment (ChrisJefferson) [inline src/debug.c]: Bah, I really don't like this, but I agree it fixes a problem and I don't have a better answer.

GCC has a special option to fix this, `-fkeep-inline-functions`, but it seems clang doesn't support that, or anything similar.

Comment (fingolfin) [reply src/debug.c]: In addition to not being portable (and there are more compilers than `gcc` and `clang` too, and hopefully more again in the future), the other problem with `-fkeep-inline-functions` is that it actually leads to *tons* of copies of these functions being emitted... Anyway, the above is perhaps not "nice", but simply works and is portable. We could even write a script to update the array for us.

**Pull request merged**